### PR TITLE
Deeper model: 5L/512d and 6L/512d with learnable PE

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

The current 4L/512d/8h architecture was established in early rounds (PR #4, chihiro) before STRING-sep learnable PE was available. With `--learnable-pe` providing a substantially richer positional representation, the model may now be depth-limited: additional attention+MLP steps could better route the richer per-point embeddings toward physics-meaningful features.

Senku PR #14 (on the old yi codebase) showed that 6L/256d outperformed 4L/512d — suggesting depth is parameter-efficient. Now that 512d is the canonical width and STRING-sep PE is available, we test whether **5L/512d** and **6L/512d** improve upon 4L/512d+STRING-sep. Each additional layer adds ~25% FLOPs vs 4L, so the runtime budget is still dominated by I/O and DDP scatter — total steps per epoch change minimally.

## Instructions

Run two arms with `torchrun --standalone --nproc_per_node=4 train.py` (DDP-4). Include `--wandb_group thorfinn-round35-depth`.

**Arm A — 5L/512d (primary arm)**
```bash
cd target/
torchrun --standalone --nproc_per_node=4 train.py \
  --agent thorfinn \
  --wandb_group thorfinn-round35-depth \
  --optimizer lion \
  --lr 1e-4 \
  --weight-decay 5e-4 \
  --clip-grad-norm 0.5 \
  --lr-warmup-epochs 1 \
  --ema-decay 0.999 \
  --batch-size 4 \
  --train-surface-points 65536 \
  --eval-surface-points 65536 \
  --train-volume-points 65536 \
  --eval-volume-points 65536 \
  --model-layers 5 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --model-slices 128 \
  --learnable-pe \
  --validation-every 1 \
  --epochs 50
```

**Arm B — 6L/512d (deeper arm)**
```bash
cd target/
torchrun --standalone --nproc_per_node=4 train.py \
  --agent thorfinn \
  --wandb_group thorfinn-round35-depth \
  --optimizer lion \
  --lr 1e-4 \
  --weight-decay 5e-4 \
  --clip-grad-norm 0.5 \
  --lr-warmup-epochs 1 \
  --ema-decay 0.999 \
  --batch-size 4 \
  --train-surface-points 65536 \
  --eval-surface-points 65536 \
  --train-volume-points 65536 \
  --eval-volume-points 65536 \
  --model-layers 6 \
  --model-hidden-dim 512 \
  --model-heads 8 \
  --model-slices 128 \
  --learnable-pe \
  --validation-every 1 \
  --epochs 50
```

Both arms use `--learnable-pe` (STRING-sep PE) and all other Lion hyperparameters from the current SOTA config. The implicit control is frieren PR #539 (4L/512d, lr=1e-4, `--learnable-pe`). These deeper models will have more parameters (~19M for 5L, ~25M for 6L vs ~13M for 4L) — if VRAM is tight, they should still fit since the bottleneck is the 65536-point point cloud, not the model weights.

## Baseline

- **Merge bar:** `val_abupt = 9.032%` (PR #517, askeladd, W&B run `brat65z4`, Lion lr=1e-4 clip=0.5, no STRING-sep PE)
- **Expected bar once frieren PR #539 lands:** ~8.1–8.5%
- **yi SOTA surface_pressure:** 5.702% | **wall_shear:** 10.257% | **volume_pressure:** 5.075%
- **Baseline reproduce command:**
```bash
cd target/
torchrun --standalone --nproc_per_node=4 train.py \
  --agent askeladd \
  --optimizer lion \
  --lr 1e-4 \
  --weight-decay 5e-4 \
  --clip-grad-norm 0.5 \
  --lr-warmup-epochs 1 \
  --ema-decay 0.999 \
  --batch-size 4 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --validation-every 1 \
  --epochs 50
```

## Reporting

Please report final val_abupt, surface_pressure_rel_l2_pct, wall_shear_rel_l2_pct, volume_pressure_rel_l2_pct, and all wall_shear_x/y/z_rel_l2_pct for both arms. Note VRAM usage per GPU. If 6L OOMs, try 6L with `--model-hidden-dim 384` as a fallback and note in results. Suggested follow-ups based on what you observe are welcome.

### CODE REVIEW

**Comment by morganmcg1** [comment]
STUDENT thorfinn:

Picking up the 5L/6L depth arms. Prior 5L attempt (run `6ofp3f5q`) crashed at end of epoch 1 with a torch._inductor InductorError (`AssertionError` in `tiling_utils.py:analyze_memory_coalescing`) on the recompile triggered by the variable-size last batch. This is a torch.compile pessimization that also seems to be making per-iter time ~3x slower than the no-compile baseline path used by askeladd PR #517.

Re-launching both arms with `--no-compile-model` (matches the PR #517 baseline). All other hyperparameters from the PR spec are kept exactly. Will report back with results.

**Comment by morganmcg1** [comment]
ADVISOR: Closing this PR — a previous advisor loop already added a closing comment but the PR was not formally closed. All of these approaches were reviewed and did not beat the current baseline (val_abupt 9.032%). Freeing this student for a new assignment.
